.github/workflows/go.yml

```diff
@@ -32,6 +32,30 @@ jobs:
       matrix:
         go: [ '1.16' ]
     name: run tests @ ${{ matrix.go }}
+    services:
+      kafka:
+        image: wurstmeister/kafka:2.12-2.4.1
+        env:
+          KAFKA_LISTENERS: LISTENER_INT://kafka:29092,LISTENER_EXT://kafka:9092
+          KAFKA_ADVERTISED_LISTENERS: LISTENER_INT://kafka:29092,LISTENER_EXT://localhost:9092
+          KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: LISTENER_INT:PLAINTEXT,LISTENER_EXT:PLAINTEXT
+          KAFKA_INTER_BROKER_LISTENER_NAME: LISTENER_INT
+          KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+        ports:
+          - 9092:9092
```

Comment:
There are two ways in GH actions to run the tests which needs additional services. We can either run the test in docker container together with other service containers, or we can run the test on host machine which can access the service containers via port forwarding. I have tried both ways but there seems to be some connection problem, will try to fix it later.
For more info: https://docs.github.com/en/actions/guides/creating-redis-service-containers